draft-packages/form/KaizenDraft/Form/RadioGroup/RadioGroup.tsx

```diff
@@ -8,6 +8,7 @@ import styles from "./styles.scss"
 export type RadioGroupProps = {
   automationId?: string
   labelText: string | React.ReactNode
+  labelId?: string
```

Comment:
I'm not a huge fan of making this a prop, particularly an optional prop, for a couple of reasons:
* Engineers won't care about this if it doesn't have any obvious effect on the implementation. It's too easy for people to miss or not care about. Ideally we want to make it impossible for people to create something that isn't accessible, or at the very least making something inaccessible should have the most friction.
* It won't retrospectively add this improvement to our existing usages, and it's very unlikely that engineers will touch old usages to add this improvement if they don't have to.

Did you explore generating a unique ID within the component with `uuid` at all? That's what I would go for, so that engineers don't need to worry about this and it's all just handled for them under the hood.

Reply:
We didn't consider `uuid`, but I did suggest adding `id` (for the container) and using it for the label, but the others said as consumers they prefer complete control over the label (I forget what the example given was - @hqta?).

Personally I would've preferred if we had refactored it to use `fieldset` and `legend`, but we were short on time, so we figured not introducing a breaking change might be the better option first? 

Perhaps I'm not understanding the intent of the pairing sessions yet. Are we meant to aim to get it to the most ideal state (because we didn't actually make it usable for @lisatassone's use case), or just find a fix for the immediate problem without breaking changes (in this case, slightly improve accessibility)? Maybe you and I can jump on a quick chat :)

Reply:
I'll hazard a guess that @hqta is going to say he might want to control the id for testing purposes? If that's the case, we could build on what we have here to generate a unique ID in the case that no `labelId` is provided, so that we get the best of both worlds.

The intent of the pairing sessions can be flexible, but generally it is to make a quick improvement or bug fix to the Kaizen codebase. Ideally it should only take 1 x 1 hour session, but sometimes we underestimate things and they need a follow up session, or to be broken out of being an advocate task. Whether it requires a breaking change or not isn't really a concern, and I'd favour getting to (or working towards) an ideal state over getting something done in the hour. With that said, making Radio Group work for Lisa's use case wasn't the intent for your session, the intent for your pairing session was to fix https://github.com/cultureamp/kaizen-design-system/issues/2409.

Reply:
Read my mind on the `id` point @dougmacknz - I like the `uuid` suggestion for improved DX. Will keep this in mind in the future.

Reply:
Awesome. Great idea. I'm on board with the fallback option. I might give this a look (also to try and refactor it) on Friday since it's our P/L day.

Reply:
I think defaulting to generating a UUID is a good improvement and just something we missed, so happy to add it. I think there's value to be able to override it and that the prop is useful (such as testing).

I was looking for ways to have more control over what I could do with this component for my use case. My approach when doing the new slice is to try and use as much Kaizen as possible and see where the gaps are and whether they are legitimate (as in, don't belong in Kaizen). 

We know that Perform's inputs are currently with design because there is a gap there that should be addressed. 

Kaizen currently dictates that all radio groups need to look a certain way because its using the Label component, but yet offers no way to enforce using the RadioField component. It doesn't accept a prop of options where it renders them out to actually create a group of radio inputs that look like what you see in the story. It stops short. Being so opinionated about the label seemed odd to me then. I'd honestly have it do that or not be there at all as the expectation that you would want from it, you don't get. Could just be me though 🤷‍♀️ 

If the idea was to have it be composable, we may not have the right abstraction.